highlight.io/pages/docs/[[...doc]].tsx

```diff
@@ -326,6 +326,34 @@ interface TocEntry {
 	children: TocEntry[]
 }
 
+// Filter quickStartContent to only include the keys that are needed for the current doc
+function getFilteredQuickStartContent(
+	newContent: string,
+	quickStartContent: any,
+) {
+	const regex = /\{(\w+(?:\["[^"]+"\])+)\}/
+	const match = newContent.match(regex)
+
+	if (match) {
+		const keyString = match[1]
+		const quickStartContentMatches = keyString
+			.split(/\["|\"]/)
+			.filter((key) => key && key !== 'quickStartContent')
+
+		const getFilteredValue = (obj: any, keys: string[]) =>
+			keys.reduce((acc, key) => acc?.[key] ?? null, obj)
+
+		const filteredQuickStartContent = quickStartContentMatches.reduceRight(
+			(obj, key) => ({ [key]: obj }),
+			getFilteredValue(quickStartContent, quickStartContentMatches),
+		)
+
+		return filteredQuickStartContent
+	} else {
+		return quickStartContent
+	}
+}
```

Comment:
looks like this works, nice! couldn't you also `JSON.parse(newContent)` and just build the object based on the keys that are required? that might be a bit more robust and simpler

Reply:
since `newContent` is in the format of 
```
<QuickStart content={quickStartContent["client"]["js"]["react"]}/>
```
I believe `JSON.parse` in this context would give an error?

Reply:
ohh my bad, you're using the regex to capture `quickStartContent["client"]["js"]["react"]` and then using the `getFilteredValue` and reduction to get the filtered object. makes sense!